fiftyone/brain/internal/core/uniqueness.py

```diff
@@ -139,26 +151,30 @@ def compute_uniqueness(
     logger.info("Uniqueness computation complete")
 
 
-def _compute_uniqueness(embeddings, metric="euclidean"):
-    K = 3
-
-    num_embeddings = len(embeddings)
-    if num_embeddings <= K:
-        return [1] * num_embeddings
-    elif num_embeddings <= _MAX_PRECOMPUTE_DISTS:
-        embeddings = skm.pairwise_distances(embeddings, metric=metric)
-        metric = "precomputed"
-    else:
-        logger.info(
-            "Computing neighbors for %d embeddings; this may take awhile...",
-            num_embeddings,
+def _compute_uniqueness(
+    embeddings, similarity_index=None, metric="euclidean", n_neighbors=3
+):
+    if similarity_index:
+        _, dists_list = similarity_index._kneighbors(
+            query=embeddings, k=n_neighbors + 1, return_dists=True
         )
+        dists = np.array(dists_list)
+    else:
+        num_embeddings = len(embeddings)
```

Comment:
This `else` block is very old; it was written before `SklearnSimilarityIndex` existed.

Probably it can just be deleted as `SklearnSimilarityIndex._kneighbors(..., return_dists=True)` should do the same thing? 🤞

Reply:
Removed else block. Default similarity index used when neither `similarity_index` or `similarity_backend` is set.